src/node_api_jsrt.cc

```diff
@@ -241,23 +241,49 @@ class ExternalCallback {
   void* _data;
 };
 
+inline napi_status JsCopyUtf8Bytes(JsValueRef key,
+                                   char** buffer, size_t* copiedBytesCount) {
+  int strLength = 0;
+  CHECK_JSRT(JsGetStringLength(key, &strLength), napi_string_expected);
+
+  // assume string contains ascii characters only
+  *buffer = reinterpret_cast<char*>(malloc(strLength + 1));
+
+  size_t written = 0;
+  size_t actualLength = 0;
+  CHECK_JSRT(JsCopyString(key, *buffer, strLength, &written, &actualLength), napi_string_expected);
+  assert(strLength == written);
+
+  // if string contains unicode characters, take slow path
+  if (actualLength != written) {
+
+    // free previously allocated buffer
+    free(*buffer);
+
+    *buffer = reinterpret_cast<char*>(malloc(actualLength + 1));
```

Comment:
yep. I will re-work this a bit and update the PR. For now, I will send another PR that unblocks the build break because of signature change.